File: src/Chapter-6-C99/Questions/Question5.c
//Write a program that asks the user to enter a two-digit number, then prints
//the number with its digits reverse. A session with the program should have
//the following appearance:
//Enter a two-digit number: 28
//The reversal is: 82;
#include <stdio.h>

int main(void)
{
    int num;
    printf("Enter a number: ");
    scanf("%d", &num);
    printf("The reversal is: ");
    //use do loop as ensures one loop
    do {
        //Print remainder
        printf("%d", num%10);
        //Reduce by 10
        num /= 10;
        //test
    } while (num > 0);

    return 0;
}

File: src/Chapter-6-Rust/Questions/Question5.rs
//Write a program that asks the user to enter a two-digit number, then prints
//the number with its digits reverse. A session with the program should have
//the following appearance:
//Enter a two-digit number: 28
//The reversal is: 82;
#[macro_use] extern crate text_io;
use std::process::Command;
//Need this to use the print!
use std::io::{self, Write};

fn main()
{
	let mut num:i32;
	print!("Enter a number: ");
	io::stdout().flush().unwrap();
	scan!("{}", num);
	print!("The reversal is: ");
	io::stdout().flush().unwrap();
	loop
	{
		//print remainder
		print!("{}", num%10);
		io::stdout().flush().unwrap();
		//Reduce by 10
		num /= 10;
		if num == 0
		{
			break;
		}
	}
	println!("");
	let _ = Command::new("cmd.exe").arg("/c").arg("pause").status();
}
